Assets/Scripts/WorldEngine/Cultures/CellCulture.cs

```diff
@@ -687,24 +687,25 @@ public float MinimumKnowledgeProgressLevel()
 
         foreach (CellCulturalKnowledge knowledge in _knowledges.Values)
         {
-//#if DEBUG
-//            if ((Manager.RegisterDebugEvent != null) && (Manager.TracingData.Priority <= 0))
-//            {
-//                if (Group.Id == Manager.TracingData.GroupId)
-//                {
-//                    SaveLoadTest.DebugMessage debugMessage = new SaveLoadTest.DebugMessage(
-//                        "CellCulture.MinimumKnowledgeProgressLevel - knowledge.Id:" + knowledge.Id + ", Group.Id:" + Group.Id,
-//                        "CurrentDate: " + Group.World.CurrentDate +
-//                        ", knowledge.IsPresent: " + knowledge.IsPresent +
-//                        //", knowledge.WasPresent: " + knowledge.WasPresent +
-//                        "");
-
-//                    Manager.RegisterDebugEvent("DebugMessage", debugMessage);
-//                }
-//            }
-//#endif
+            //#if DEBUG
+            //            if ((Manager.RegisterDebugEvent != null) && (Manager.TracingData.Priority <= 0))
+            //            {
+            //                if (Group.Id == Manager.TracingData.GroupId)
+            //                {
+            //                    SaveLoadTest.DebugMessage debugMessage = new SaveLoadTest.DebugMessage(
+            //                        "CellCulture.MinimumKnowledgeProgressLevel - knowledge.Id:" + knowledge.Id + ", Group.Id:" + Group.Id,
+            //                        "CurrentDate: " + Group.World.CurrentDate +
+            //                        ", knowledge.IsPresent: " + knowledge.IsPresent +
+            //                        //", knowledge.WasPresent: " + knowledge.WasPresent +
+            //                        "");
+
+            //                    Manager.RegisterDebugEvent("DebugMessage", debugMessage);
+            //                }
+            //            }
+            //#endif
 
-            float level = knowledge.CalculateExpectedProgressLevel();
+            // if progress level equals 0 that means the knowledge can't really progress. So we ignore it
+            float level = (knowledge.ProgressLevel > 0) ? knowledge.CalculateExpectedProgressLevel() : 1;
```

Comment:
So the entire fix was just to add this statement? That's pretty nice.
How do knowledges get stuck at 0 progress level?

Reply:
That's a deeper issue that I need to investigate further. One possible cause is that groups acquire a knowledge from a neighbor but do not have the necessary conditions to sustain it so the knowledge drops to 0. But not all knowledges have a removal mechanism so they linger at 0 forever. This is bad because any active knowledge requires to be updated, which means wasted CPU cycles for knowledges that have no way to progress.